server.js

```diff
@@ -1,15 +1,191 @@
+    await newDogChat.save();
+    res.redirect("/cat-room");
+  }
+});
+
+//READ - grabs the cat room chats and posts them on the cat room page
+app.get("/api/cat-chat", async (req, res) => {
+  let catChats = await CatChat.find({});
+  res.send(catChats);
+});
+
+//CREATE - enables user to create dog room chats
+app.post("/dog-chat", async (req, res) => {
+  let userObj = req.body;
+  req.body.timestamp = timeStamp();
+  //catches messages over 500 characters
+  if (userObj.message.length > 500) {
+    res.status(403).redirect("/dog-room");
+  }
+  if (userObj.username === `` || userObj.message === ``) {
+    res.status(403).redirect("/dog-room");
+  } else {
+    let newDogChat = new DogChat(userObj);
+    await newDogChat.save();
+    res.redirect("/dog-room");
+  }
+});
+
+//READ - grabs the dog room chats and posts them on the dog room page
+app.get("/api/dog-chat", async (req, res) => {
+  let dogChats = await DogChat.find({});
+  res.send(dogChats);
+});
+
+//CREATE - enables user to create main chat room chats
+app.post("/main-chat", async (req, res) => {
+  let userObj = req.body;
+  req.body.timestamp = timeStamp();
+  //catches messages over 500 characters
+  if (userObj.message.length > 500) {
+    res.status(403).redirect("/");
+  }
+  //catches empty username/message blocks
+  if (userObj.username === `` || userObj.message === ``) {
+    res.status(403).redirect("/");
+  } else {
+    let newDogChat = new Chat(userObj);
+    await newDogChat.save();
+    res.redirect("/");
+  }
+});
+
+//READ - grabs the main chat room chats and posts them on the main page
+app.get("/api/main-chat", async (req, res) => {
+  let allChats = await Chat.find({});
+  res.send(allChats);
+});
+
+//sets a 10 second timeout on the chat
+setTimeout(() => {}, 10000);
 
+//timestamp function
+function timeStamp() {
+  //creates new Date object at time
+  let date = new Date();
+  //gets Hour item from Date object
+  let hour = date.getHours();
+  //initiates AM/PM variable
+  let amPM;
+  //changes 24 hour time into 12 hour time, sets AM/PM accordingly
+  if (hour > 12) {
+    //Subtract 12 to do away with military time
+    hour = `${hour - 12}`;
+    //And puts PM on time stamp to denote it is after Noon
+    amPM = `PM`;
+    //If the hour equals noon exactly
+  } else if (hour === 12) {
+    //Set the am pm variable to PM
+    amPM = `PM`;
+  } else {
+    //Otherwise set the hour as is to AM
+    amPM = `AM`;
+  }
+  //get Minute item from Date object
+  let minute = date.getMinutes();
+  //handles '00'-'09' times
+  if (minute < 10) {
+    //Have the minute with a 0 in front (i.e. :06)
+    minute = `0${minute}`;
+  }
+  //sets clock
+  let time = `${hour}:${minute}${amPM}`;
+  //sets mm/dd/yyyy
+  let stamp = `${date.getMonth()}/${date.getDate()}/${date.getFullYear()}`;
+  //sets full timestamp
+  let timeStamp = `(${time} ${stamp})`;
+  //sends timestamp
+  return timeStamp;
+}
 
+//Catch-all error route
+app.get("*", (req, res) => {
```

Comment:
Good idea, you need to connect it to your React front end to have it work though